internal/enginecache/cache.go

```diff
@@ -237,29 +237,109 @@ func ValidateOrUpdateEngine(targetOS, cachePath, requiredEngineVersion string, m
 
 	log.Printf("Downloading engine for platform %s at version %s...", EngineConfig(targetOS, mode), requiredEngineVersion)
 
-	file := ""
-	switch targetOS {
-	case "linux":
-		file += "linux"
-	case "darwin":
-		file += "macosx"
-	case "windows":
-		file += "windows"
-	}
-	file += fmt.Sprintf("_x64-host_%s.zip", mode.Name)
-	engineDownloadURL := fmt.Sprintf("https://github.com/flutter-rs/engine-builds/releases/download/f-%s/%s", requiredEngineVersion, file)
+	if mode == build.DebugMode {
+		targetedDomain := "https://storage.googleapis.com"
```

Comment:
Didn't we have a function that gives the correct storage URI based on whether the user is behind CN firewall?

Reply:
Only for flutter, becase they provide such service. I readded it below